### PR TITLE
updated contract calls page

### PR DESCRIPTION
**Pull request by andrei-marinica**
#### Description of the pull request (what is new / what has changed)

The contract calls needed to be brought up to date and explained in greater detail.

Also noticed we had no documentation on `#[payable(...)]` and direct transfers. I had to write it.

Testing framework docs are not in scope, will come in next PR.

#### Did you test the changes locally ?
- [X] yes
- [ ] no

#### Which category (categories) does this pull request belong to?
- [x] document new feature
- [x] update documentation that is not relevant anymore
- [x] add examples or more information about a component
- [ ] fix grammar issues
- [ ] other

image

### CODE REVIEW

File: docs/developers/developer-reference/sc-payments.md [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+
+The following methods will perform a transfer-execute call. This is a type of asynchronous call that does not provide a callback, or any other feedback from the receiver. For direct transferring of funds, this call type is ideal. 
+
+- `self.send().direct_egld(to, amount)` sends EGLD directly to an address.
+- `self.send().direct_non_zero_egld(to, amount)` sends EGLD directly to an address, if the amount is non-zero. Does nothing otherwise.
+- `self.send().direct(to, token, nonce, amount)` sends EGLD or a single ESDT token directly to an address. The `token` argument is of type `EgldOrEsdtTokenIdentifier`.
+- `self.send().direct_non_zero(to, token, nonce, amount)` sends EGLD or a single ESDT token directly to an address, if the amount is non-zero. Does nothing otherwise.
+- `self.send().direct_esdt(to, token, nonce, amount)` sends a single ESDT token directly to an address.
+- `self.send().direct_non_zero_esdt_payment(to, payment)` sends a single ESDT token directly to an address, if the amount is non-zero. Does nothing otherwise.
+- `self.send().direct_multi(to, payments)` sends one or more ESDT tokens to destination. The `payments` argument is a list of such payments.
```
Comment (ovidiuolteanu) [inline docs/developers/developer-reference/sc-payments.md:114]:
```suggestion
- `self.send().direct_multi(to, payments)` sends one or more ESDT tokens to destination. The `payments` argument is a list of such payments. If at least one of the amounts is zero, execution will fail.
```

File: docs/developers/developer-reference/sc-payments.md [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+
+[comment]: # (mx-context-auto)
+
+### Sending payments to contract endpoints
+
+Sending tokens to a contract endpoint is a contract call, and we have a [type-safe and clean way of doing this](/developers/developer-reference/sc-contract-calls).
+
+Even if you do not know what the endpoint or the arguments will be at compile time, we still recommend creating a `ContractCall` first, and then decorating and sending it, as in [these examples](/developers/developer-reference/sc-contract-calls#no-proxy).
+
+There are, howver, a few direct API methods. Just to mention them:
```
Comment (ovidiuolteanu) [inline docs/developers/developer-reference/sc-payments.md:130]:
```suggestion
However, a few direct API methods exist. Just to mention them:
```

**Review by ovidiuolteanu — CHANGES_REQUESTED** [review]

File: docs/developers/developer-reference/sc-contract-calls.md [hunk 3]
```diff
@@ -183,59 +500,56 @@ fn caller_endpoint(&self) {
...
-#[payable("*")]
-#[endpoint(myEndpoint)]
-fn my_payable_endpoint(&self, arg: BigUint) -> BigUint {
-	let payment = self.call_value().egld_or_single_esdt();
-	// implementation
-}
+Promises (or multi-async calls) are a new feature that will be introduced in mainnet release 1.6. They are very similar to the old asynchronous calls, with the difference that launching them does not terminate current execution, and there can be several launched from the same transaction.
+
+:::caution
+Because this feature is currently not available on mainnet, contracts need to enable the "promises" feature flag in `Cargo.toml` to use the functionality:
```
Comment (ovidiuolteanu) [inline docs/developers/developer-reference/sc-contract-calls.md:519]: Don't know what would be the best solution here to avoid the need to modify the docs after the release of 1.6

File: docs/developers/developer-reference/sc-contract-calls.md [hunk 4]
```diff
@@ -252,74 +566,253 @@ fn my_endpoint_callback(&self, #[call_result] result: ManagedAsyncCallResult<Big
...
+- We need to specify the gas for the call, because the execution of our transaction will continue and it needs to know how much gas it can keep.
+- We need to specify the amount of gas for the callback. This is the exact amount of gas reserved for the callback, irrespective of how much the target contract consumes.
+
+
+
+[comment]: # (mx-context-auto)
+
+### Transfer-execute
+
+Transfer-execute calls are similar to asynchronous calls, but they can have no callback, and thus the caller cannot react in any way to what happens with the callee.
```
Comment (ovidiuolteanu) [inline docs/developers/developer-reference/sc-contract-calls.md:582]:
```suggestion
Transfer-execute calls are similar to asynchronous calls, but they can have no callback, thus the caller cannot react in any way to what happens with the callee.
```

File: docs/developers/developer-reference/sc-contract-calls.md [hunk 5]
```diff
@@ -252,74 +566,253 @@ fn my_endpoint_callback(&self, #[call_result] result: ManagedAsyncCallResult<Big
...
+
+Just like promises, there can be multiple such calls launched from a transaction, but unlike promises, they are alreay available on mainnet.
```
Comment (ovidiuolteanu) [inline docs/developers/developer-reference/sc-contract-calls.md:584]:
```suggestion
Just like promises, there can be multiple such calls launched from a transaction, but unlike promises, these are already available on mainnet.
```
Same question here regarding the need to modify afterwards.

File: docs/developers/developer-reference/sc-contract-calls.md [hunk 6]
```diff
@@ -252,74 +566,253 @@ fn my_endpoint_callback(&self, #[call_result] result: ManagedAsyncCallResult<Big
...
+
+Transfer-execute calls do not need any further configuration (other than explicit an explicit gas limit), and therefore there is no specific object type associated with them. They can be launched immediately:
```
Comment (ovidiuolteanu) [inline docs/developers/developer-reference/sc-contract-calls.md:586]:
```suggestion
Transfer-execute calls do not need any further configuration (other than an explicit gas limit), therefore there is no specific object type associated with them. They can be launched immediately:
```

File: docs/developers/developer-reference/sc-contract-calls.md [hunk 7]
```diff
@@ -252,74 +566,253 @@ fn my_endpoint_callback(&self, #[call_result] result: ManagedAsyncCallResult<Big
...
-		.with_callback(self.callbacks().my_endpoint_callback())
-		.call_and_exit();
-}
+
+
+[comment]: # (mx-context-auto)
+
+### Synchronous calls
+
+Synchronous calls are executed inline: this means execution is interrupted while they are executed, and resumed afterwards. We also get the result of the execution right away and we can use it immediately in the transaction.
```
Comment (ovidiuolteanu) [inline docs/developers/developer-reference/sc-contract-calls.md:601]:
```suggestion
Synchronous calls are executed inline: this means execution is interrupted while they are executed and resumed afterwards. We also get the result of the execution right away and we can use it immediately in the transaction.
```

File: docs/developers/developer-reference/sc-contract-calls.md [hunk 8]
```diff
@@ -252,74 +566,253 @@ fn my_endpoint_callback(&self, #[call_result] result: ManagedAsyncCallResult<Big
...
+
+The catch is that synchronous calls can only be sent to contracts in the __same shard__ as the caller. They will fail otherwise.
```
Comment (ovidiuolteanu) [inline docs/developers/developer-reference/sc-contract-calls.md:603]:
```suggestion
:::caution
Synchronous calls can only be sent to contracts in the __same shard__ as the caller. They will fail otherwise.
:::
```

**Review by ovidiuolteanu — CHANGES_REQUESTED** [review]

File: docs/developers/developer-reference/sc-contract-calls.md [hunk 9]
```diff
@@ -51,38 +104,269 @@ fn my_endpoint(&self, arg: BigUint) -> BigUint {
...
-After performing this call, you can execute some more code in the caller contract, using `biguint_result` as you wish.
 
-NOTE: Keep in mind that this only works for same-shard contracts. If the contracts are in different shards, you have to use async-calls or transfer-and-execute.
+[comment]: # (mx-context-auto)
+
+### No proxy
+
+The point of the proxies is to help us build contract calls in a type-safe manner. But this is by no means compulsory. Sometimes we specifically want to build contract calls by hand and serialize the arguments ourselves.
+
+The point is to create a `ContractCallNoPayment` object. We'll discuss how to add the payments later.
```
Comment (CostinCarabas) [inline docs/developers/developer-reference/sc-contract-calls.md:170]: In order to do that, we want to create a [...]

File: docs/developers/developer-reference/sc-contract-calls.md [hunk 10]
```diff
@@ -51,38 +104,269 @@ fn my_endpoint(&self, arg: BigUint) -> BigUint {
...
+The `ContractCallNoPayment` has two type arguments: the API and the expected return type. If we are in a contract, the API will always be the same as for the entire contract. To avoid having to explicitly specify it, we can use the following syntax:
+
+```rust
+let mut contract_call = self.send()
+	.contract_call::<ResultType>(to, endpoint_name);
+contract_call.push_raw_argument(arg1_encoded);
+contract_call.push_raw_argument(arg2_encoded);
+```
+
+If we are not in a smart contract we might do the same thing using the same, equivalent syntax:
```
[the remark this answers is not in the document]
Comment (CostinCarabas) [reply docs/developers/developer-reference/sc-contract-calls.md:181]: For me it's a bit ambiguous. Can you rephrase?

File: docs/developers/developer-reference/sc-contract-calls.md [hunk 11]
```diff
@@ -51,38 +104,269 @@ fn my_endpoint(&self, arg: BigUint) -> BigUint {
...
+### Diagram
+
+Up to here we have created a contract call without payment, so an object of type `ContractCallNoPayment`, in the following ways:
+
+```mermaid
+graph LR
+    gen-proxy[Generated Proxy] --> ccnp[ContractCallNoPayment]
+    man-proxy[Manual Proxy] --> ccnp
+    ccnp-new["ContractCallNoPayment::new(to, function)"] --> ccnp
+```
```
Comment (CostinCarabas) [inline docs/developers/developer-reference/sc-contract-calls.md:203]: Can you center the image?

File: docs/developers/developer-reference/sc-contract-calls.md [hunk 12]
```diff
@@ -252,74 +566,253 @@ fn my_endpoint_callback(&self, #[call_result] result: ManagedAsyncCallResult<Big
...
+    cc ---->|".deploy_contract()
+              .deploy_from_source()"| exec-deploy["⚙️ Deploy contract"]
+    exec-deploy --> result[Requested Result]
+    cc ---->|".deploy_contract()
+              .deploy_from_source()"| exec-upg["⚙️ Upgrade contract"]
+```
+
+The object encoding these calls is called `ContractDeploy`. Unlike the contract calls, there is a single such object.
+
+Creating this object is done in a similar fashion: either via proxies, or by hand. Constructors in proxies naturally produce `ContractDeploy` objects:
```
Comment (ovidiuolteanu) [inline docs/developers/developer-reference/sc-contract-calls.md:725]:
```suggestion
Creating this object is done in a similar fashion: either via proxies, or manually. Constructors in proxies naturally produce `ContractDeploy` objects:
```

File: docs/developers/developer-reference/sc-contract-calls.md [hunk 13]
```diff
@@ -252,74 +566,253 @@ fn my_endpoint_callback(&self, #[call_result] result: ManagedAsyncCallResult<Big
...
+:::
+
+The methods for executing contract deploys are as follows:
+- `.deploy_contract(code, code_metadata)` - deploys a new contract with the code given by the contract.
+- `.deploy_from_source(source_address, code_metadata)` - deploys a new contract with the same code as the code of the contract at `source_address`. The advantage is that the contract doesn't need to handle the new contract code, which could be quite a large data blob. This saves gas. It requires that we have the code already deployed somewhere else.
+
+
+[comment]: # (mx-context-auto)
+
+### Upgrade
```
Comment (ovidiuolteanu) [inline docs/developers/developer-reference/sc-contract-calls.md:801]: In the next mainnet release we'll have deffered inits for upgrades. We'll have to care for this as well very soon. Maybe I more places.

**Review by ovidiuolteanu — CHANGES_REQUESTED** [review]

File: docs/developers/developer-reference/sc-contract-calls.md [hunk 14]
```diff
@@ -51,38 +104,269 @@ fn my_endpoint(&self, arg: BigUint) -> BigUint {
...
+## Contract calls: gas
+
+Specifying gas is fairly straightforward. All contract call objects have a `with_gas_limit` method, so gas can be specified at any point.
+
+Not all contract calls require explicit specification of the gas limit, leaving it out is sometimes fine. Notably:
+- async calls will halt execution and consume all the remaining gas, so specifying the gas limit is not necessary for them;
+- synchronous calls will by default simply use all the available gas as the upper limit, since unspent gas is returned to the caller anyway.
+
+On the other hand,promises and transfer-execute calls do require gas to be explicitly specified.
+
```
Comment (CostinCarabas) [inline docs/developers/developer-reference/sc-contract-calls.md:339]: Add snippets of code with `with_gas_limit` as an example.

File: docs/developers/developer-reference/sc-contract-calls.md [hunk 15]
```diff
@@ -51,38 +104,269 @@ fn my_endpoint(&self, arg: BigUint) -> BigUint {
...
 
 [comment]: # (mx-context-auto)
 
-## Types of Contract to Contract calls
+## Contract calls: execution
+
+There are several ways in which contract calls are launched from another contract. Currently they are:
+- asynchronous calls:
+	- single asynchronous calls:
+	- promises (multi ple asynchronous calls),
```
Comment (CostinCarabas) [inline docs/developers/developer-reference/sc-contract-calls.md:350]: `multi ple` -> multiple

File: docs/developers/developer-reference/sc-contract-calls.md [hunk 16]
```diff
@@ -99,7 +383,10 @@ fn caller_endpoint(&self) {
 
 ### Callbacks
 
-If you want to perform some logic based on the result of the async call, or just some cleanup after the call, you have to declare a callback function. For example, let's say we want to do something based if the result is even, and something else if the result is odd, and do some cleanup in case of error. Our callback function would look something like this:
+But before sending the asynchronous call, we most likely will want to also configure a callback for it. This is the only way that our contract will be able to react to the outcome of its execution.
+
+Let's imagine that `my_endpoint` returns a `BigUint`, and we want to do something if the result is even, and something else if the result is odd. We also want to do some cleanup in case of error. Our callback function would look something like this:
```
Comment (CostinCarabas) [inline docs/developers/developer-reference/sc-contract-calls.md:388]: `my_endpoint` -> `caller_endpoint`

File: docs/developers/developer-reference/sc-contract-calls.md [hunk 17]
```diff
@@ -136,13 +426,40 @@ fn caller_endpoint(&self) {
...
 
-### Callback Arguments
+### The callback closure
+
+Assume there is some additional context that we want to pass from our contract directly to the callback. We cannot rely on the called contract to do the job for us, we want to be in full control of this context.
+
+This context forms the contents of our callback closure.
 
-Your callback may have additional arguments that are given to it at the time of launching the async call. These will be automatically saved before performing the initial async call, and they will be retrieved when the callback is called. Example:
+More specifically, all callback arguments other than the `#[call_result]` will be passed to it before launching the call, they will be saved by the framework somewhere in the contract storage, then given to the callback and deleted.
```
Comment (CostinCarabas) [inline docs/developers/developer-reference/sc-contract-calls.md:460]: remove `somewhere`

File: docs/developers/developer-reference/sc-contract-calls.md [hunk 18]
```diff
@@ -183,59 +500,56 @@ fn caller_endpoint(&self) {
...
 
-Let's say you want to call a `#[payable]` endpoint, with this definition:
-```rust
-#[payable("*")]
-#[endpoint(myEndpoint)]
-fn my_payable_endpoint(&self, arg: BigUint) -> BigUint {
-	let payment = self.call_value().egld_or_single_esdt();
-	// implementation
-}
+Promises (or multi-async calls) are a new feature that will be introduced in mainnet release 1.6. They are very similar to the old asynchronous calls, with the difference that launching them does not terminate current execution, and there can be several launched from the same transaction.
```
Comment (CostinCarabas) [inline docs/developers/developer-reference/sc-contract-calls.md:516]: ~are a feature that are available from the mainnet release 1.6.~

Comment (CostinCarabas) [reply docs/developers/developer-reference/sc-contract-calls.md:516]: Also, a suggestion: programmers (technical people) like bullets. You can create 2 bullets here.

File: docs/developers/developer-reference/sc-contract-calls.md (hunk 7 above)
Comment (CostinCarabas) [reply docs/developers/developer-reference/sc-contract-calls.md:601]: I would add caller and callee to the explaination.
`this means the caller execution is interrupted while the callee function is executed`

File: docs/developers/developer-reference/sc-contract-calls.md [hunk 19]
```diff
@@ -252,74 +566,253 @@ fn my_endpoint_callback(&self, #[call_result] result: ManagedAsyncCallResult<Big
...
+self.contract_proxy(callee_sc_address)
+	.init(my_biguint_arg)
+
 ```
 
-Where `dest_sc_address` is the address of the callee contract, and `endpoint_name` would be `b"myEndpoint"`.
+```rust
+self.contract_proxy()
+	.init(my_biguint_arg)
+	.with_egld_transfer(egld_amount)
```
Comment (CostinCarabas) [inline docs/developers/developer-reference/sc-contract-calls.md:761]: is `with_egld_transfer` available on deploy? I think this is an upgrade, right?

Comment (andrei-marinica) [reply docs/developers/developer-reference/sc-contract-calls.md:761]: It is!

File: docs/developers/developer-reference/sc-contract-calls.md [hunk 20]
```diff
@@ -51,38 +104,269 @@ fn my_endpoint(&self, arg: BigUint) -> BigUint {
...
+Importing a smart contract crate only works if both contracts use the exact framework version. Otherwise, the compiler will (rightfully) complain that the interfaces do not perfectly match.
+
+In case the target contract is not under our control, it is often wiser to just write the proxy of interest by hand.
+:::
+
+
+
+[comment]: # (mx-context-auto)
+
+### Manually specified proxies
```
[the remark this answers is not in the document]
Comment (andrei-marinica) [reply docs/developers/developer-reference/sc-contract-calls.md:128]: Not sure if composed is the better word here. It suggests smaller bits were put together, but these proxies are written from scratch.

File: docs/developers/developer-reference/sc-contract-calls.md (hunk 13 above)
Comment (andrei-marinica) [reply docs/developers/developer-reference/sc-contract-calls.md:801]: I know ....
But first we need the implementation, then the docs ....

File: docs/developers/developer-reference/sc-contract-calls.md [hunk 21]
```diff
@@ -1,45 +1,98 @@
...
+```rust
+adder_proxy.add(3u32)
+```
+
+Here, we have a proxy to the adder contract. The `add` method doesn't call the smart contract directly. Instead, it produces a contract call object that contains the data field `add@03`, which is something that the blockchain can make sense of. We will see later how this contract call can end up actually being executed. But until then, let's see how we can get hold of one of these proxies.
+
 This guide provides some examples on how to call a contract from another contract. More examples can be found in [the contract composability feature tests](https://github.com/multiversx/mx-sdk-rs/tree/master/contracts/feature-tests/composability).
 
 There are three ways of doing these calls:
 - importing the callee contract's source code and using the auto-generated proxy (recommended)
```
[the remark this answers is not in the document]
Comment (andrei-marinica) [reply docs/developers/developer-reference/sc-contract-calls.md:48]: A new way of creating proxies is coming soon, will be the recommended way. Writing them by hand is not necessarily the way.

File: docs/developers/developer-reference/sc-contract-calls.md (hunk 3 above)
Comment (andrei-marinica) [reply docs/developers/developer-reference/sc-contract-calls.md:519]: We will modify them.

**Review by ovstinga — DISMISSED** [review]

File: docs/developers/developer-reference/sc-contract-calls.md [hunk 22]
```diff
@@ -1,95 +1,387 @@
...
+
+## Contract calls: base
+
+Smart contract calls at the blockchain level have no notion of arity, or data types. That is, the blockchain itself does not validate the number of arguments (or results), and each of these only appear as raw binary data fields.
+
+It is the contract that keeps track of the number of arguments, and deserializes them. If a transaction has the wrong number of arguments, it is only the contract itself that will be able to complain. If the types are off, it is only during deserialization that the contract will know.
+
+The description of a smart contract's inputs is known as the [ABI](/developers/data/abi), and lives off-chain. In short, the ABI is a collection of endpoint names, with argument names and type descriptions. To be able to effectively call a smart contract, it is useful to know its ABI.
+
+We call this contract call helper a __proxy__. All it does is that it provides a typed interface to any Rust program, it takes the typed arguments and it serializes them according to the [MultiversX serialization format](/developers/data/serialization-overview).
```
Comment (camilbancioiu) [inline docs/developers/developer-reference/sc-contract-calls.md:35]: For extra clarity:

```suggestion
Knowing the ABI of a contract enables us to call it cleanly. The ABI is used to create a __proxy__ for the contract, which is a helper component that stands in for the contract in our code. All it does is that it provides a typed interface to any Rust program, it takes the typed arguments and it serializes them according to the [MultiversX serialization format](/developers/data/serialization-overview).
```

File: docs/developers/developer-reference/sc-contract-calls.md [hunk 23]
```diff
@@ -1,95 +1,387 @@
...
+This function doesn't do much, it just tries to sort out the proxy trait imports, and neatly initializes the proxy for you.
+
 This function creates an object that contains all the endpoints of the callee contract, and it handles the serialization automatically.
 
 Let's say you have the following endpoint in the contract you wish to call:
 
 ```rust
 #[endpoint(myEndpoint)]
-fn my_endpoint(&self, arg: BigUint) -> BigUint {
+fn callee_endpoint(&self, arg: BigUint) -> BigUint {
```
Comment (CostinCarabas) [inline docs/developers/developer-reference/sc-contract-calls.md:102]: Also, modify the name of the endpoint

File: docs/developers/developer-reference/sc-contract-calls.md [hunk 24]
```diff
@@ -1,95 +1,387 @@
...
 
-If you use the rust-analyser VSCode extension, it might complain that it can't find this, but if you actually build the contract, the compiler can find it just fine.
+If you use the [rust-analyser VSCode extension](https://marketplace.visualstudio.com/items?itemName=rust-lang.rust-analyzer), it might complain that it can't find this, but if you actually build the contract, the compiler can find it just fine.
 
-Once you've imported the contract and any external modules it might use, you have to declare your proxy creator function:
+Once you've imported the contract and any external modules it might use, you have to declare a proxy creator function in the contract:
 ```rust
 #[proxy]
-fn contract_proxy(&self, sc_address: ManagedAddress) -> contract_namespace::Proxy<Self::Api>;
+fn callee_contract_proxy(&self, sc_address: ManagedAddress) -> contract_namespace::Proxy<Self::Api>;
```
Comment (CostinCarabas) [inline docs/developers/developer-reference/sc-contract-calls.md:91]: Below you used the proxy with `callee_sc_address`. Please rename here also

**Review by CostinCarabas — DISMISSED** [review]

**Review by ovstinga — APPROVED** [review]

File: docs/developers/developer-reference/sc-contract-calls.md [hunk 25]
```diff
@@ -122,31 +417,61 @@ fn my_endpoint_callback(
...
-	// other code here
+	// previous code here
 
-	self.contract_proxy(callee_sc_address)
-		.my_endpoint(my_biguint_arg)
+	self.callee_contract_proxy(callee_sc_address)
+		.callee_endpoint(my_biguint_arg)
 		.async_call()
-		.with_callback(self.callbacks().my_endpoint_callback())
+		.with_callback(self.callbacks().callee_endpoint_callback())
```
Comment (camilbancioiu) [inline docs/developers/developer-reference/sc-contract-calls.md:432]: `callee_endpoint_callback` is called here, should it rather be a function pointer? Without the parantheses?

File: docs/developers/developer-reference/sc-payments.md [hunk 26]
```diff
@@ -0,0 +1,143 @@
...
+
+
+We have seen how contracts can accomodate receiving tokens. Sending them is, in principle, even more straightforward, as it only involves calling methods from the `self.send()` component.
+
+
+[comment]: # (mx-context-auto)
+
+### Sending payments directly
+
+Contracts can send tokens directly, without calling any endpoint at the destination, by using several methods from the API.
```
Comment (camilbancioiu) [inline docs/developers/developer-reference/sc-payments.md:104]: If the contract is payable by default, otherwise it's disallowed.

File: docs/developers/developer-reference/sc-contract-calls.md (hunk 25 above)
Comment (andrei-marinica) [reply docs/developers/developer-reference/sc-contract-calls.md:432]: I wish we had the power to redefine function pointers. We don't.

The next best thing: fooling the developer. This is no the actual callback, it is the callback proxy, that function actually produces a `CallbackCall`.

I won't be writing about it today. Maybe another day.

Marvel in the mischief!

**Review by CostinCarabas — DISMISSED** [review]

File: docs/developers/developer-reference/sc-contract-calls.md (hunk 22 above)
Comment (andrei-marinica) [reply docs/developers/developer-reference/sc-contract-calls.md:35]: Not completely correct. The proxy right now is not created based on the ABI.

File: docs/developers/developer-reference/sc-payments.md (hunk 26 above)
Comment (andrei-marinica) [reply docs/developers/developer-reference/sc-payments.md:104]: This is about sending, not receiving. We talked about direct transfers and payable contracts in a different section.

File: docs/developers/developer-reference/sc-contract-calls.md (hunk 22 above)
Comment (andrei-marinica) [reply docs/developers/developer-reference/sc-contract-calls.md:35]: I rephrased it a little differently.

**Review by ovstinga — APPROVED** [review]

**Review by camilbancioiu — APPROVED** [review]
